src/main/java/org/folio/edgecommonspring/util/ApiKeyHelper.java

```diff
@@ -0,0 +1,77 @@
+package org.folio.edgecommonspring.util;
+
+import static org.folio.edge.api.utils.Constants.HEADER_API_KEY;
+import static org.folio.edge.api.utils.Constants.PARAM_API_KEY;
+import static org.folio.edge.api.utils.Constants.PATH_API_KEY;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.PostConstruct;
+import javax.servlet.ServletRequest;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ApiKeyHelper {
```

Comment:
Just curious, the most of current logic is common for both edge-common and edge-common-spring. Maybe it make sense to move at least getFromParam, getFromPath, getFromHeader methods to common library edge-api-utils?

Reply:
`ApiKeyHelper` in edge-common use another context - `io.vertx.ext.web.RoutingContext`
For spring way we use `ServletRequest`

You can check here - https://github.com/folio-org/edge-common/blob/master/src/main/java/org/folio/edge/core/ApiKeyHelper.java

Reply:
Ok, agree, but getFromParam, getFromPath, getFromHeader methods?

Reply:
We could make it an abstract class.... 

implement most of this, including the enum and `public String getEdgeApiKey(Object request)` here, and make these abstract:

```
String getFromParam(Object request)
String getFromHeader(Object request)
String getFromPath(Object request)
```